task20/js/main.js

```diff
@@ -0,0 +1,95 @@
+import '../css/style.css'
+import { createElementWithClassName } from './utils/createElement'
+import { renderLoading } from './modules/loading'
+import { removeLoading } from './modules/loading'
+import { renderErrorMessage } from './modules/error'
+
+const endpoint = "https://mocki.io/v1/159d72ba-bd6b-429f-a28c-713eeaba9f72";
+
+const parent = document.getElementById("js-parent");
+
+const getFetchData = async () => {
+    const response = await fetch(endpoint);
+    // const response = await fetch("https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236");//空配列
+
+    if (!response.ok){
+        console.error(`$(response:status):$(response:statusText)`);
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    }
+    return await response.json();
+};
+
+const fetchTableData = () => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(getFetchData(endpoint));
+        }, 3000);
+        // setTimeout(() => {
+        //     resolve(getFetchData("https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236"));
+        // }, 3000);//空配列
+    });
+};
```

Comment:
nits
For an endpoint, if it is only used here, it may not need to be included in the argument.

```suggestion
const getFetchData = async (endpoint) => {
    const response = await fetch(endpoint);
    // const response = await fetch("https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236");//空配列

    if (!response.ok){
        console.error(`$(response:status):$(response:statusText)`);
        renderErrorMessage(parent, "問題が発生し表示することができません。");
    }
    return await response.json();
};

const fetchTableData = () => {
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve(getFetchData("https://mocki.io/v1/159d72ba-bd6b-429f-a28c-713eeaba9f72"));
        }, 3000);
        // setTimeout(() => {
        //     resolve(getFetchData("https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236"));
        // }, 3000);//空配列
    });
};
```

Reply:
I felt the code Chihiro-san gave me was more readable, so I installed arguments and eliminated variables.
[77c81cf](https://github.com/seeeen044/JavaScript_handson/pull/21/commits/77c81cfd884c2b3ac9c5dbe9fef3ab107b0554ae)